File: convert-csv-to-json.js
const fs = require('fs');
const path = require('path');

const csvPath = path.join('c:', 'Users', 'i.ulises', 'Desktop', 'tablaDias.csv');
const jsonPath = path.join('c:', 'Users', 'i.ulises', 'Desktop', 'tablaDias.json');

const csv = fs.readFileSync(csvPath, 'utf8');
const lines = csv.trim().split(/\r?\n/);
const headers = lines[0].split(';');

const data = lines.slice(1)
  .filter(line => line.trim())
  .map(line => {
    const values = line.split(';');
    const obj = {};
    headers.forEach((h, i) => {
      const val = values[i] !== undefined ? values[i].trim() : '';
      obj[h] = val === '' || isNaN(Number(val)) ? val : Number(val);
    });
    return obj;
  });

fs.writeFileSync(jsonPath, JSON.stringify(data, null, 2), 'utf8');
console.log('Convertido:', data.length, 'filas ->', jsonPath);

File: convert_csv_to_json.py
# -*- coding: utf-8 -*-
import csv
import json

csv_path = r'c:\Users\i.ulises\Desktop\tablaDias.csv'
json_path = r'c:\Users\i.ulises\Desktop\tablaDias.json'

data = []
with open(csv_path, 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f, delimiter=';')
    for row in reader:
        for key in row:
            val = row[key].strip()
            if val and val.isdigit():
                row[key] = int(val)
        data.append(row)

with open(json_path, 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)

print(f'Convertido: {len(data)} filas -> {json_path}')
